collections/map_test.go

```diff
@@ -0,0 +1,47 @@
+package collections
+
+import (
+	storetypes "github.com/cosmos/cosmos-sdk/store/types"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestMap(t *testing.T) {
+	sk, ctx := deps()
+	m := NewMap(sk, NewPrefix("hi"), Uint64Key, Uint64Value)
```

Comment:
I actually liked the previous API, where you just passed in an int. I wonder if we could have the prefix type be parameterized at the Schema level once we can align on that

Reply:
this API accepts both int, string and []byte

Reply:
I preferred the enforcement on `uint8` as prefix too... but backwards compat...

Reply:
Yes.

Reply:
Let's address in a follow-up